core/parallel_state_processor.go

```diff
@@ -95,6 +95,66 @@ type ExecutionTask struct {
 	blockContext vm.BlockContext
 }
 
+// NewWarmExecTasks creates a list of ExecutionTask objects for warming up the state reader cache.
+func NewWarmExecTasks(bc *BlockChain, chainCfg *params.ChainConfig, header *types.Header, baseState *state.StateDB, coinbase common.Address, txs []*types.Transaction, evmCfg vm.Config) ([]blockstm.ExecTask, error) {
+	tasks := make([]blockstm.ExecTask, 0, len(txs))
+	if len(txs) == 0 {
+		return tasks, nil
+	}
+
+	finalState := baseState.Copy()
+	var (
+		receipts types.Receipts
+		allLogs  []*types.Log
+		usedGas  = new(uint64)
+	)
+	// Create block and EVM block context
+	blockNumber := header.Number
+	blockHash := header.Hash()
+	blockTime := header.Time
+	blockCtx := NewEVMBlockContext(header, bc, &coinbase)
+	shouldDelayFeeCal := false
+
+	for i, tx := range txs {
```

Comment:
There could be large amount of transactions in `txs`, which may cause unnecessary warm up. For example, if there are 10k pending txs, but only 20% of them will be included in the next block, 80% of compute would be wasted. One way of limiting is to include transactions until the sum of their gas limit reaches the block gas limit.

Reply:
Correct but I think the only way to accurately determine the gas used is to execute the txs and this is a time consuming process itself.